docs for single-host devfile endpoins exposure on subdomains (#1617)
codenvy/che-docs: modules/installation-guide/partials/con_workspace-exposure-strategies.adoc

This is convenient for TLS-secured {prod-short} servers because it is sufficient to have a single certificate for the {prod-short} server, which will cover all the workspace component deployments as well.

> There are two types of endpoints, that are defined in the devfiles, exposures in single-host strategy. To configured it, use `++CHE_INFRA_KUBERNETES_SINGLEHOST_WORKSPACE_DEVFILE__ENDPOINT__EXPOSURE++`. It is effective only in single-host server strategy..

Reviewer: I'm not sure if English uses the "implanted" sentences the same way as Czech. For me, the English version is hard to follow unless I translate it into Czech at which point it suddenly makes much more sense :)

What about rewording it as:

There are 2 ways of exposing the endpoints specified in the devfile. These can be configured using the `CHE_INFRA_KUBERNETES_SINGLEHOST_WORKSPACE_DEVFILE__ENDPOINT__EXPOSURE` environment variable of the `{prod-short}`. This environment variable is only effective with the single-host server strategy and is applicable to all workspaces of all users.

Author: yes, fixed

=== devfile endpoints: `single-host`

`++CHE_INFRA_KUBERNETES_SINGLEHOST_WORKSPACE_DEVFILE__ENDPOINT__EXPOSURE: 'single-host'++`

> This single-host configuration exposes the endpoints on subpaths, e.g.: `++https://<che-host>/serverihzmuqqc/go-cli-server-8080++`. This limits the exposed components and user applications. Any absolute URL generated on the server side that points back to the server does not work. This is because the server is hidden behind a path-rewriting Ingress that hides the workspace and the component-specific URL prefix from the server.

Reviewer: I'd replace "Ingress" with "reverse proxy", because that's what makes the endpoint not work.

Also, "that hides the workspace and the component-specific URL prefix from the server." - the "server" here is confusing, because we're talking about components and user applications in the prior sentence. I would probably say "that hides the unique URL path prefix from the component or user application."

Author: this is just copy-paste of what was already there. good time to refresh it. fixed.

> This single-host configuration exposes the endpoints on subpaths, e.g.: `++https://<che-host>/serverihzmuqqc/go-cli-server-8080++`. This limits the exposed components and user applications. Any absolute URL generated on the server side that points back to the server does not work. This is because the server is hidden behind a path-rewriting reverse proxy that hides the unique URL path prefix from the component or user application.

Reviewer 2: ```suggestion
This single-host configuration exposes the endpoints on subpaths, for example: `++https://<che-host>/serverihzmuqqc/go-cli-server-8080++`. This limits the exposed components and user applications. Any absolute URL generated on the server side that points back to the server does not work. This is because the server is hidden behind a path-rewriting reverse proxy that hides the unique URL path prefix from the component or user application.
```

For example, when the user accesses the hypothetical `[{prod-url}/component-prefix-djh3d/app/index.php]` URL, the application sees the request coming to `++https://internal-host/app/index.php++`. If the application used the host in the URL that it generates in its UI, it would not work because the internal host is different from the externally visible host. However, if the application used an absolute path as the URL (for the example above, this would be `/app/index.php`), such URL would still not work. This is because on the outside, such URL does not point to the application, because it is missing the component-specific prefix.

Therefore, only applications that use relative URLs in their UI work with the single-host workspace exposure strategy.

=== devfile endpoints: `multi-host`
`++CHE_INFRA_KUBERNETES_SINGLEHOST_WORKSPACE_DEVFILE__ENDPOINT__EXPOSURE: 'multi-host'++`

> This single-host configuration exposes the endpoints on subdomains, e.g.: `++http://serverihzmuqqc-go-cli-server-8080.<che-host>++`. These endpoints will be exposed on unsecured HTTP port. We're using dedicated Ingress or Route for such endpoints, even with `gateway` single-host setup.

Reviewer: This single-host configuration exposes the endpoints on subdomains, e.g.: `++http://serverihzmuqqc-go-cli-server-8080.<che-host>++`. These endpoints are be exposed on an unsecured HTTP port. A dedicated Ingress or Route is used for such endpoints, even with `gateway` single-host setup.

Author: fixed

> This single-host configuration exposes the endpoints on subdomains, e.g.: `++http://serverihzmuqqc-go-cli-server-8080.<che-host>++`. These endpoints are exposed on an unsecured HTTP port. A dedicated Ingress or Route is used for such endpoints, even with `gateway` single-host setup.

Reviewer 2: ```suggestion
This single-host configuration exposes the endpoints on subdomains, for example: `++http://serverihzmuqqc-go-cli-server-8080.<che-host>++`. These endpoints are exposed on an unsecured HTTP port. A dedicated Ingress or Route is used for such endpoints, even with `gateway` single-host setup.
```

> This configuration limits the usability of previews shown directly in the editor page when {prod-short} is configured with TLS. Because `https` pages allows communication only with secured endpoints, users must open their application previews on another tab of the browser.

Reviewer: "in another browser tab".

Author: fixed

> This configuration limits the usability of previews shown directly in the editor page when {prod-short} is configured with TLS. Because `https` pages allows communication only with secured endpoints, users must open their application previews in another browser tab.

Reviewer 2: ```suggestion
This configuration limits the usability of previews shown directly in the editor page when {prod-short} is configured with TLS. Since `https` pages allows communication only with secured endpoints, users must open their application previews in another browser tab.
```

> This configuration limits the usability of previews shown directly in the editor page when {prod-short} is configured with TLS. Because `https` pages allows communication only with secured endpoints, users must open their application previews in another browser tab.

Reviewer 2: ```suggestion
This configuration limits the usability of previews shown directly in the editor page when {prod-short} is configured with TLS. Because `https` pages allow communication only with secured endpoints, users must open their application previews in another browser tab.
```

> This configuration limits the usability of previews shown directly in the editor page when {prod-short} is configured with TLS. Because `https` pages allow communication only with secured endpoints, users must open their application previews in another browser tab.

Author: ```suggestion
This configuration limits the usability of previews shown directly in the editor page when {prod-short} is configured with TLS. Since `https` pages allow communication only with secured endpoints, users must open their application previews in another browser tab.
```
